src/Core/Services/Cache/DabCacheService.cs

```diff
@@ -0,0 +1,137 @@
+                       httpContext: _httpContextAccessor.HttpContext!,
+                       args: null,
+                       dataSourceName: queryMetadata.DataSource);
+
+                   ctx.Options.SetSize(EstimateCacheEntrySize(cacheKey: cacheKey, cacheValue: result?.ToString()));
+                   ctx.Options.SetDuration(duration: TimeSpan.FromSeconds(cacheEntryTtl));
+
+                   return result;
+               });
+
+        return result;
+    }
+
+    /// <summary>
+    /// Creates a cache key using the request metadata resolved from a built SqlQueryStructure
+    /// Format: DataSourceName:QueryText:JSON_QueryParameters
+    /// Example: 7a07f92a-1aa2-4e2a-81d6-b9af0a25bbb6:select * from MyTable where id = @param1 = :{"@param1":{"Value":"42","DbType":null}}
+    /// </summary>
+    /// <returns>Cache key string</returns>
+    private string CreateCacheKey(DatabaseQueryMetadata queryMetadata)
+    {
+        StringBuilder cacheKeyBuilder = new();
+        cacheKeyBuilder.Append(queryMetadata.DataSource);
+        cacheKeyBuilder.Append(KEY_DELIMITER);
+        cacheKeyBuilder.Append(queryMetadata.QueryText);
+        cacheKeyBuilder.Append(KEY_DELIMITER);
+        cacheKeyBuilder.Append(JsonSerializer.Serialize(queryMetadata.QueryParameters));
+        string cacheKey = cacheKeyBuilder.ToString();
+
+        if (_logger?.IsEnabled(LogLevel.Trace) ?? false)
+        {
+            _logger.LogTrace(message: "{cacheKey}", cacheKey);
+        }
+
+        return cacheKeyBuilder.ToString();
+    }
+
+    /// <summary>
+    /// Estimates the size of the cache entry in bytes.
+    /// The cache entry is the concatenation of the cache key and cache value.
+    /// </summary>
+    /// <param name="cacheKey">Cache key string.</param>
+    /// <param name="cacheValue">Cache value as a serialized JSON payload.</param>
+    /// <returns>Size in bytes.</returns>
+    /// <exception cref="ArgumentException">Thrown when the cacheKey value is empty or whitespace.</exception>
+    /// <exception cref="OverflowException">Thrown when the cache entry size is too big.</exception>
+    private long EstimateCacheEntrySize(string cacheKey, string? cacheValue)
+    {
+        if (string.IsNullOrWhiteSpace(cacheKey))
+        {
+            throw new ArgumentException(message: "Cache key should not be empty.");
+        }
+
+        try
+        {
+            checked
+            {
+                long size = 0L;
+                long cacheValueSize = string.IsNullOrWhiteSpace(cacheValue) ? 0L : cacheValue.Length;
+                size += cacheKey.Length * sizeof(char);
+                size += cacheValueSize * sizeof(char);
+                return size;
+            }
+        }
+        catch (OverflowException)
+        {
+            if (_logger?.IsEnabled(LogLevel.Trace) ?? false)
+            {
+                _logger.LogTrace(message: "Cache entry is too big.");
```

Comment:
The cache key for which the entry is big can be logged. Could be helpful for investigating/debugging

Reply:
Cache key logging can be enabled with appsettings.json configuration to change logging level for fusion cache.